src/v/cloud_storage/cache_service.cc

```diff
@@ -75,14 +75,16 @@ cache::cache(
   config::binding<double> disk_reservation,
   config::binding<uint64_t> max_bytes_cfg,
   config::binding<std::optional<double>> max_percent,
-  config::binding<uint32_t> max_objects) noexcept
+  config::binding<uint32_t> max_objects,
+  config::binding<uint16_t> walk_concurrency) noexcept
```

Comment:
nit: why do you need binding if you never subscribing to its notifications?

Reply:
I always read it when starting the walk so it is live-settable without subscribing to notifications.

I should actually mark it as not requiring restart.